server/src/main/java/io/crate/plugin/HttpTransportPlugin.java

```diff
@@ -1,116 +0,0 @@
- * with Crate these terms will supersede the license and you may use the
- * software solely pursuant to the terms of the relevant commercial agreement.
- */
-
-package io.crate.plugin;
-
-import io.crate.protocols.http.CrateNettyHttpServerTransport;
-import org.elasticsearch.client.Client;
-import org.elasticsearch.client.node.NodeClient;
-import org.elasticsearch.cluster.service.ClusterService;
-import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
-import org.elasticsearch.common.network.NetworkService;
-import org.elasticsearch.common.settings.Setting;
-import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.common.util.BigArrays;
-import org.elasticsearch.common.xcontent.NamedXContentRegistry;
-import org.elasticsearch.env.Environment;
-import org.elasticsearch.env.NodeEnvironment;
-import org.elasticsearch.http.HttpServerTransport;
-import org.elasticsearch.indices.breaker.CircuitBreakerService;
-import org.elasticsearch.plugins.ActionPlugin;
-import org.elasticsearch.plugins.NetworkPlugin;
-import org.elasticsearch.plugins.Plugin;
-import org.elasticsearch.threadpool.ThreadPool;
-
-import java.util.Collection;
-import java.util.Collections;
-import java.util.List;
-import java.util.Map;
-import java.util.function.Supplier;
-
-import static org.elasticsearch.common.network.NetworkModule.HTTP_TYPE_KEY;
-import static org.elasticsearch.http.HttpTransportSettings.SETTING_HTTP_COMPRESSION;
-
-
-public class HttpTransportPlugin extends Plugin implements NetworkPlugin, ActionPlugin {
-
-    private static final String CRATE_HTTP_TRANSPORT_NAME = "crate";
-
-    private final PipelineRegistry pipelineRegistry;
-
-    public HttpTransportPlugin(Settings settings) {
-        this.pipelineRegistry = new PipelineRegistry(settings);
-    }
-
-    public String name() {
-        return "http";
-    }
-
-    public String description() {
-        return "Plugin for extending HTTP transport";
-    }
-
-    @Override
-    public Collection<Object> createComponents(Client client,
-                                               ClusterService clusterService,
-                                               ThreadPool threadPool,
-                                               NamedXContentRegistry xContentRegistry,
-                                               Environment environment,
-                                               NodeEnvironment nodeEnvironment,
-                                               NamedWriteableRegistry namedWriteableRegistry) {
-        // pipelineRegistry is returned here so that it's bound in guice and can be injected in other places
-        return Collections.singletonList(pipelineRegistry);
-    }
-
-    @Override
-    public List<Setting<?>> getSettings() {
-        return Collections.emptyList();
-    }
-
-    @Override
-    public Settings additionalSettings() {
-        return Settings.builder()
-            .put(HTTP_TYPE_KEY, CRATE_HTTP_TRANSPORT_NAME)
```

Comment:
looks good to me 👍 

this setting was not used?

Reply:
It is set here:

https://github.com/crate/crate/pull/9946/files#diff-ce1c2ef0997d0c94964f483274f72f4aR100